os/linux/weston/plugin/compositor-iahwc.c

```diff
@@ -1420,6 +1421,8 @@ static void session_notify(struct wl_listener *listener, void *data) {
     weston_log("activating session\n");
     weston_compositor_wake(compositor);
     weston_compositor_damage_all(compositor);
+    //FIXME: Output isn't initialized to anything yet we are trying to use its
+    //spin_lock member
```

Comment:
Good catch! the lock and unlock should be inside the for_each loop.
@kalyankondapally Is this correct?

Reply:
hmm I dont understand the issue here,

Reply:
We are doing a 
`lock(&output->spin_lock);`
but output is defined only inside 
`wl_list_for_each(output, &compositor->output_list, base.link) {`

Reply:
hmm I see, makes sense to move into the loop ya

Reply:
@kalyankondapally An error is raised by cppcheck about this